packages/pluggableWidgets/line-chart-native/src/utils/SeriesDataLoadingFunction.ts

```diff
@@ -0,0 +1,203 @@
+
+export function loadSeries(series: SeriesType[]): LineChartSeries[] | null {
+    const loadedSeries: LineChartSeries[] = [];
+
+    for (const element of series) {
+        if (element.type === "static") {
+            const result = loadStaticSeries(element);
+
+            if (!result) {
+                return null;
+            }
+
+            loadedSeries.push(result);
+        } else {
+            const result = loadDynamicSeries(element);
+
+            if (!result) {
+                return null;
+            }
+
+            loadedSeries.push(...result);
+        }
+    }
+
+    return loadedSeries;
+}
+
+function loadStaticSeries(series: SeriesType): LineChartSeries | null {
+    const { interpolation, staticLineStyle, staticSeriesName, staticStylePropertyName, type } = series;
+
+    if (type !== "static") {
+        throw Error("Expected series to be static");
+    }
+
+    if (staticSeriesName && !staticSeriesName.value && typeof staticSeriesName.value !== "string") {
+        return null;
+    }
+
+    const dataPoints = extractDataPoints(series);
+
+    if (!dataPoints) {
+        return null;
+    }
+
+    return {
+        dataPoints,
+        interpolation,
+        lineStyle: staticLineStyle,
+        name: staticSeriesName ? staticSeriesName.value : undefined,
+        stylePropertyName: staticStylePropertyName
+    };
+}
+
+function loadDynamicSeries(series: SeriesType): LineChartSeries[] | null {
+    const { dynamicLineStyle, interpolation, type } = series;
+
+    if (type !== "dynamic") {
+        throw Error("Expected series to be dynamic");
+    }
+
+    const dataSourceItemGroups = groupDataSourceItems(series);
+
+    if (!dataSourceItemGroups) {
+        return null;
+    }
+
+    const loadedSeries: LineChartSeries[] = [];
+
+    for (const itemGroup of dataSourceItemGroups) {
+        const dataPoints = extractDataPoints(series, itemGroup.items);
+
+        if (!dataPoints) {
+            return null;
+        }
+
+        loadedSeries.push({
+            dataPoints,
+            interpolation,
+            lineStyle: dynamicLineStyle,
+            name: itemGroup.dynamicSeriesNameValue,
+            stylePropertyName: itemGroup.dynamicStylePropertyNameValue
+        });
+    }
+
+    return loadedSeries;
+}
+
+function groupDataSourceItems(series: SeriesType): DataSourceItemGroup[] | null {
+    const { dynamicDataSource, dynamicSeriesName, dynamicStylePropertyName, groupByAttribute, type } = series;
+
+    if (type !== "dynamic") {
+        throw Error("Expected series to be dynamic");
+    }
+
+    const dataSource = ensure(dynamicDataSource);
+
+    if (!dataSource.items) {
+        return null;
+    }
+
+    const dataSourceItemGroupsResult: DataSourceItemGroup[] = [];
+
+    for (const item of dataSource.items) {
+        const groupByAttributeValue = ensure(groupByAttribute)(item);
+
+        if (
+            !groupByAttributeValue.value &&
+            typeof groupByAttributeValue.value !== "string" &&
```

Comment:
do we really need to add the typing check? You are already testing if the value is undefined, 0, empty or null at line 120

Reply:
It should not return null when there is an empty string, basically only when it's undefined. I changed the check to an undefined check only.